ZeroWasteFriendly/Scenes/Library/LibraryController.swift

```diff
@@ -56,6 +68,13 @@ extension LibraryController: LibraryPresentable {
         postsDataSource = posts
         tableView.reloadData()
     }
+
+    func animateCarouselHeight(height: CGFloat) {
+        UIView.animate(withDuration: 0.25) {
+            self.collectionViewHeight.constant = height
+            self.view.layoutSubviews()
```

Comment:
in most of cases changing HEIGHT of ui looks bad, it's better to push the window outside the visible view
make the tableview PUSH collection view out of view

keep height static = 200
keep your constraint: colView.top = safeArea.top - but lower priority
make new constraint and add IBOutlet - colview.bottom = supervioew.top = 0
try in xib if toggling this constraint works
make an outlet (STRONG reference, because WEAK + isActive = false deinitializes it)

animate only isActive = true/false

Reply:
![Mar-09-2020 07-50-43](https://user-images.githubusercontent.com/32341779/76189360-c2f7b580-61da-11ea-84d5-03871c7cea64.gif)

this is something I've achieved- but something is not working properly. Layout is breaking. Something with your flow layout method